fix(components): [tour] ensure `current` prop is controlled when passed

DDDDD12138 opened the pull request:
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

fix: #19667
fix: #24008

### Description
Refactored the handling of the `current` prop to ensure it is controlled when passed.
> For example, if not using `v-model:current` and only passing a fixed value `:current='1'`, the tour should always stay at step 1.

### Preview
[BUG Preview](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIHR5cGU9XCJwcmltYXJ5XCIgQGNsaWNrPVwib3BlbiA9IHRydWVcIj5CZWdpbiBUb3VyPC9lbC1idXR0b24+XG5cbiAgPGVsLWRpdmlkZXIgLz5cblxuICA8ZWwtc3BhY2U+XG4gICAgPGVsLWJ1dHRvbiBndWlkZT1cImJ0bjFcIiBAY2xpY2s9XCJuZXh0XCI+Y2xpY2sgaGVyZTwvZWwtYnV0dG9uPlxuICAgIDxlbC1kaWFsb2cgaWQ9XCJkaWFsb2dcIiB2LW1vZGVsPVwidmlzaWJsZVwiIHRpdGxlPVwidGl0bGVcIj5cbiAgICAgIDxkaXY+Y29udGVudDwvZGl2PlxuICAgIDwvZWwtZGlhbG9nPlxuICA8L2VsLXNwYWNlPlxuXG4gIDxlbC10b3VyIHYtbW9kZWw9XCJvcGVuXCIgOmN1cnJlbnQ9XCJjdXJyZW50XCIgOnotaW5kZXg9XCIzMDAwXCI+XG4gICAgPGVsLXRvdXItc3RlcCB0YXJnZXQ9XCJbZ3VpZGU9YnRuMV1cIiB0aXRsZT1cIlVwbG9hZCBGaWxlXCIgZGVzY3JpcHRpb249XCJQdXQgeW91IGZpbGVzIGhlcmUuXCJcbiAgICAgIDpuZXh0LWJ1dHRvbi1wcm9wcz1cIm5leHRCdXR0b25Qcm9wc1wiIC8+XG4gICAgPGVsLXRvdXItc3RlcCB0YXJnZXQ9XCIjZGlhbG9nXCIgdGl0bGU9XCJkaWFsb2dcIiBkZXNjcmlwdGlvbj1cInRoaXMgaXMgZGlhbG9nXCIgLz5cbiAgPC9lbC10b3VyPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3Qgb3BlbiA9IHJlZihmYWxzZSlcblxuY29uc3QgY3VycmVudCA9IHJlZigwKVxuXG5jb25zdCBuZXh0QnV0dG9uUHJvcHMgPSB7XG4gIG9uQ2xpY2s6ICgpID0+IHtcbiAgICBuZXh0KClcbiAgfVxufVxuXG5jb25zdCB2aXNpYmxlID0gcmVmKGZhbHNlKVxuXG5mdW5jdGlvbiBuZXh0KCkge1xuICB2aXNpYmxlLnZhbHVlID0gdHJ1ZVxuICBzZXRUaW1lb3V0KCgpID0+IHtcbiAgICBjdXJyZW50LnZhbHVlID0gMVxuICB9LCAzMDApXG59XG48L3NjcmlwdD5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC9kaXN0L2luZGV4LmNzcycsICdodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvdGhlbWUtY2hhbGsvZGFyay9jc3MtdmFycy5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbFNldHRsZWQoc3R5bGVzKVxufVxuIiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L2Rpc3QvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L1wiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7fX0=)

[PR Preview](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIHR5cGU9XCJwcmltYXJ5XCIgQGNsaWNrPVwib3BlbiA9IHRydWVcIj5CZWdpbiBUb3VyPC9lbC1idXR0b24+XG5cbiAgPGVsLWRpdmlkZXIgLz5cblxuICA8ZWwtc3BhY2U+XG4gICAgPGVsLWJ1dHRvbiBndWlkZT1cImJ0bjFcIiBAY2xpY2s9XCJuZXh0XCI+Y2xpY2sgaGVyZTwvZWwtYnV0dG9uPlxuICAgIDxlbC1kaWFsb2cgaWQ9XCJkaWFsb2dcIiB2LW1vZGVsPVwidmlzaWJsZVwiIHRpdGxlPVwidGl0bGVcIj5cbiAgICAgIDxkaXY+Y29udGVudDwvZGl2PlxuICAgIDwvZWwtZGlhbG9nPlxuICA8L2VsLXNwYWNlPlxuXG4gIDxlbC10b3VyIHYtbW9kZWw9XCJvcGVuXCIgOmN1cnJlbnQ9XCJjdXJyZW50XCIgOnotaW5kZXg9XCIzMDAwXCI+XG4gICAgPGVsLXRvdXItc3RlcCB0YXJnZXQ9XCJbZ3VpZGU9YnRuMV1cIiB0aXRsZT1cIlVwbG9hZCBGaWxlXCIgZGVzY3JpcHRpb249XCJQdXQgeW91IGZpbGVzIGhlcmUuXCJcbiAgICAgIDpuZXh0LWJ1dHRvbi1wcm9wcz1cIm5leHRCdXR0b25Qcm9wc1wiIC8+XG4gICAgPGVsLXRvdXItc3RlcCB0YXJnZXQ9XCIjZGlhbG9nXCIgdGl0bGU9XCJkaWFsb2dcIiBkZXNjcmlwdGlvbj1cInRoaXMgaXMgZGlhbG9nXCIgLz5cbiAgPC9lbC10b3VyPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3Qgb3BlbiA9IHJlZihmYWxzZSlcblxuY29uc3QgY3VycmVudCA9IHJlZigwKVxuXG5jb25zdCBuZXh0QnV0dG9uUHJvcHMgPSB7XG4gIG9uQ2xpY2s6ICgpID0+IHtcbiAgICBuZXh0KClcbiAgfVxufVxuXG5jb25zdCB2aXNpYmxlID0gcmVmKGZhbHNlKVxuXG5mdW5jdGlvbiBuZXh0KCkge1xuICB2aXNpYmxlLnZhbHVlID0gdHJ1ZVxuICBzZXRUaW1lb3V0KCgpID0+IHtcbiAgICBjdXJyZW50LnZhbHVlID0gMVxuICB9LCAzMDApXG59XG48L3NjcmlwdD5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vcHJldmlldy0xOTY3NS1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2Rpc3QvaW5kZXguY3NzJywgJ2h0dHBzOi8vcHJldmlldy0xOTY3NS1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL3RoZW1lLWNoYWxrL2RhcmsvY3NzLXZhcnMuY3NzJ10ubWFwKChzdHlsZSkgPT4ge1xuICAgIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgICAgbGluay5ocmVmID0gc3R5bGVcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgICB9KVxuICB9KVxuICByZXR1cm4gUHJvbWlzZS5hbGxTZXR0bGVkKHN0eWxlcylcbn1cbiIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJQbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0xOTY3NS1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2Rpc3QvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy0xOTY3NS1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2Rpc3QvaW5kZXguY3NzIn19)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Tour navigation now syncs current step correctly in controlled and uncontrolled modes; advancing steps emits update events in the correct order and visible step title updates reliably.
  * Closing the tour resets the current step to the first step without emitting an extra change event.

* **Tests**
  * Added async tests covering navigation, event ordering, deferred updates, and close/reset behavior.

* **Chores**
  * Removed the built-in default for the Tour's current prop so omission is treated as uncontrolled.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

DDDDD12138 committed:
feat: ensure 'current' prop is controlled when passed

DDDDD12138 committed:
feat: add test

rzzf commented on packages/components/tour/src/tour.vue:
The default value here should also be removed.

rzzf reviewed (commented):
After removing default value, the parameters of the `change` event seem to be incorrect.. [repro](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIHR5cGU9XCJwcmltYXJ5XCIgQGNsaWNrPVwib3BlbiA9IHRydWVcIj5CZWdpbiBUb3VyPC9lbC1idXR0b24+XG5cbiAgPGVsLWRpdmlkZXIgLz5cblxuICA8ZWwtc3BhY2U+XG4gICAgPGVsLWJ1dHRvbiBndWlkZT1cImJ0bjFcIiBAY2xpY2s9XCJuZXh0XCI+Y2xpY2sgaGVyZTwvZWwtYnV0dG9uPlxuICAgIDxlbC1kaWFsb2cgaWQ9XCJkaWFsb2dcIiB2LW1vZGVsPVwidmlzaWJsZVwiIHRpdGxlPVwidGl0bGVcIj5cbiAgICAgIDxkaXY+Y29udGVudDwvZGl2PlxuICAgIDwvZWwtZGlhbG9nPlxuICA8L2VsLXNwYWNlPlxuXG4gIDxlbC10b3VyXG4gICAgdi1tb2RlbD1cIm9wZW5cIlxuICAgIDpjdXJyZW50PVwiY3VycmVudFwiXG4gICAgOnotaW5kZXg9XCIzMDAwXCJcbiAgICBAY2hhbmdlPVwiKGN1cnJlbnQpID0+IGNvbnNvbGUubG9nKCdjaGFuZ2U6ICcsIGN1cnJlbnQpXCJcbiAgPlxuICAgIDxlbC10b3VyLXN0ZXBcbiAgICAgIHRhcmdldD1cIltndWlkZT1idG4xXVwiXG4gICAgICB0aXRsZT1cIlVwbG9hZCBGaWxlXCJcbiAgICAgIGRlc2NyaXB0aW9uPVwiUHV0IHlvdSBmaWxlcyBoZXJlLlwiXG4gICAgICA6bmV4dC1idXR0b24tcHJvcHM9XCJuZXh0QnV0dG9uUHJvcHNcIlxuICAgIC8+XG4gICAgPGVsLXRvdXItc3RlcFxuICAgICAgdGFyZ2V0PVwiI2RpYWxvZ1wiXG4gICAgICB0aXRsZT1cImRpYWxvZ1wiXG4gICAgICBkZXNjcmlwdGlvbj1cInRoaXMgaXMgZGlhbG9nXCJcbiAgICAvPlxuICA8L2VsLXRvdXI+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IHNldHVwIGxhbmc9XCJ0c1wiPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBvcGVuID0gcmVmKGZhbHNlKVxuXG5jb25zdCBjdXJyZW50ID0gcmVmKDApXG5cbmNvbnN0IG5leHRCdXR0b25Qcm9wcyA9IHtcbiAgb25DbGljazogKCkgPT4ge1xuICAgIG5leHQoKVxuICB9LFxufVxuXG5jb25zdCB2aXNpYmxlID0gcmVmKGZhbHNlKVxuXG5mdW5jdGlvbiBuZXh0KCkge1xuICB2aXNpYmxlLnZhbHVlID0gdHJ1ZVxuICBzZXRUaW1lb3V0KCgpID0+IHtcbiAgICBjdXJyZW50LnZhbHVlID0gMVxuICB9LCAzMDApXG59XG48L3NjcmlwdD5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vdW5wa2cuY29tL2VsZW1lbnQtcGx1c0BsYXRlc3QvZGlzdC9pbmRleC5jc3MnLCAnaHR0cHM6Ly91bnBrZy5jb20vZWxlbWVudC1wbHVzQGxhdGVzdC90aGVtZS1jaGFsay9kYXJrL2Nzcy12YXJzLmNzcyddLm1hcCgoc3R5bGUpID0+IHtcbiAgICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICAgIGxpbmsuaHJlZiA9IHN0eWxlXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gICAgfSlcbiAgfSlcbiAgcmV0dXJuIFByb21pc2UuYWxsU2V0dGxlZChzdHlsZXMpXG59XG4iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly91bnBrZy5jb20vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL3VucGtnLmNvbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vdW5wa2cuY29tL2VsZW1lbnQtcGx1c0BsYXRlc3QvZGlzdC9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJodHRwczovL3VucGtnLmNvbS9lbGVtZW50LXBsdXNAbGF0ZXN0L1wiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL3VucGtnLmNvbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJfbyI6e319)

DDDDD12138 committed:
fix(components): [tour] correct controlled current change event

DDDDD12138 commented:
> After removing default value, the parameters of the `change` event seem to be incorrect

You're right: after removing the default value, `change` could emit a stale parameter in the controlled case because it was still triggered from the button click flow before the parent synced `current`.

I've updated the implementation to properly distinguish controlled vs uncontrolled usage, removed the `current` default fallback in `withDefaults`, and moved `change` emission to the actual `current` change path. This way, `change` is emitted only after the step really changes, and it always carries the latest value.

DDDDD12138 commented on packages/components/tour/__tests__/tour.test.tsx:
```diff
@@ -77,6 +77,124 @@ describe('Tour.vue', () => {
+    )
+    const tourStepOneComponent = wrapper.getComponent(TourStep)
+    const nextBtn = tourStepOneComponent.find('.next-btn')
+    nextBtn.trigger('click')
```
This is intentional: the click is followed by `await nextTick()` to assert the state before the later `flushPromises()` parent sync, so the async flow is already covered.

DDDDD12138 committed:
fix(components): [tour] correct controlled current change event
